command/command.go

```diff
@@ -140,12 +140,13 @@ func (e *ExecutionRequest) Execute(ctx context.Context, execution ExecutionReque
 				escapeZombie(&execution)
 			}
 		})
+		defer timer.Stop()
 	}
 
 	if err := cmd.Start(); err != nil {
 		// Something unexpected happended when attepting to
 		// fork/exec, return immediately.
-		return nil, err
+		return resp, err
```

Comment:
Usually, when a function returns `val, err`, if `err != nil` one is not expected to use `val` at all. Why return `resp, err` instead of `nil, err`?

Reply:
Hm, I feel ya. Unfortunately the agent currently expects, regardless of err, that the response is always present. Meaning we'll encounter a panic if the process fails to start.

I'll revert this change for now and we can handle it another time.

Reply:
I see. The real fix then, in my opinion, is to change the agent to properly handle the fact that when there is an error, it can't also always expect a meaningful response.